internal/service/autoscaling/group.go

```diff
@@ -2175,6 +2175,10 @@ func statusGroupCapacity(ctx context.Context, conn *autoscaling.AutoScaling, elb
 
 		for _, v := range scalingActivities {
 			if statusCode := aws.StringValue(v.StatusCode); statusCode == autoscaling.ScalingActivityStatusCodeFailed && aws.Int64Value(v.Progress) == 100 {
+				if strings.Contains(aws.StringValue(v.StatusMessage), "Invalid IAM Instance Profile") {
```

Comment:
hm, this could match errors on multiple parameters. What do you think about constraining it to the ARN: `Invalid IAM Instance Profile ARN`, so it doesn't retry for invalid instance profile names or other parameters not related to a delay on state propagation.

Reply:
Yeah, sounds reasonable

Reply:
Probably not... I recall why I didn't add the `ARN` part. It's because the autoscaling group gets the IAM instance profile from the launch template, which supports both `name` and `ARN` as identifiers: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#instance-profile